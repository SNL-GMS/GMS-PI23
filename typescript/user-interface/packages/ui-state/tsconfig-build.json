{
  "extends": "./tsconfig.json",
  "compilerOptions": {
    "noEmit": false,
    "composite": true,
    "tsBuildInfoFile": "./.cache/.tsbuildinfo",
    "rootDir": "src/ts"
  },
  "references": [
    {
      "path": "../common-model/tsconfig-build.json"
    },
    {
      "path": "../common-util/tsconfig-build.json"
    },
    {
      "path": "../ui-workers/tsconfig-build.json"
    },
    {
      "path": "../ui-util/tsconfig-build.json"
    },
    {
      "path": "../ui-wasm/tsconfig-build.json"
    },
    {
      "path": "../weavess-core/tsconfig-build.json"
    }
  ],
  "include": ["src/**/*"]
}
