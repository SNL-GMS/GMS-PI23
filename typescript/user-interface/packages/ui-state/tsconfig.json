{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
    "baseUrl": ".",
    "noEmit": true,
    "outDir": "./lib",
    "paths": {
      "@gms/common-model": ["../common-model/src/ts/common-model.ts"],
      "@gms/common-model/lib/*": ["../common-model/src/ts/*"],
      "@gms/common-util": ["../common-util/src/ts/common-util.ts"],
      "@gms/common-util/lib/*": ["../common-util/src/ts/*"],
      "@gms/ui-util": ["../ui-util/src/ts/ui-util.ts"],
      "@gms/ui-util/lib/*": ["../ui-util/src/ts/*"],
      "@gms/ui-workers": ["../ui-workers/src/ts/ui-workers.ts"],
      "@gms/ui-workers/lib/*": ["../ui-workers/src/ts/*"],
      "@gms/ui-wasm": ["../ui-wasm/src/ts/ui-wasm.ts"],
      "@gms/ui-wasm/lib/*": ["../ui-wasm/src/ts/*"],
      "@gms/weavess-core": ["../weavess-core/src/ts/weavess-core.ts"],
      "@gms/weavess-core/lib/*": ["../weavess-core/src/ts/*"]
    },
    "lib": ["ES2020", "DOM", "DOM.Iterable", "ScriptHost", "webworker"]
  },
  "include": ["src/**/*", "__tests__/**/*", "__mocks__/**/*"]
}
