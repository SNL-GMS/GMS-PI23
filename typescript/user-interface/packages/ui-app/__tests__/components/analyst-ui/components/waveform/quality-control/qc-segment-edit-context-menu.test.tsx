import { qcSegment } from '@gms/common-model/__tests__/__data__';
import { getStore } from '@gms/ui-state';
import { render, waitFor } from '@testing-library/react';
import React from 'react';
import { Provider } from 'react-redux';

import {
  QcSegmentEditContextMenu,
  QcSegmentEditContextMenuContent
} from '~analyst-ui/components/waveform/quality-control/qc-segment-edit-context-menu';
import type { QcSegmentContextMenuOpenFunc } from '~analyst-ui/components/waveform/quality-control/types';

describe('QC Segment Edit Context Menu', () => {
  it('exists', () => {
    expect(QcSegmentEditContextMenu).toBeDefined();
    expect(QcSegmentEditContextMenuContent).toBeDefined();
  });

  it('renders QcSegmentEditContextMenuContent', () => {
    // Empty content
    let container = render(<QcSegmentEditContextMenuContent qcSegment={undefined} />);
    expect(container.container).toMatchSnapshot();

    // One QC Segment
    container = render(
      <Provider store={getStore()}>
        <QcSegmentEditContextMenuContent qcSegment={qcSegment} />
      </Provider>
    );
    expect(container.container).toMatchSnapshot();
  });

  it('renders QcSegmentContextMenu', async () => {
    let qcContextMenu: QcSegmentContextMenuOpenFunc;

    const Display = function Display() {
      return (
        <QcSegmentEditContextMenu
          getOpenCallback={callback => {
            qcContextMenu = callback;
          }}
        />
      );
    };

    const container = render(<Display />);
    expect(container.container).toMatchSnapshot();

    const mouseEvent = ({
      nativeEvent: new MouseEvent('contextmenu', {
        clientX: 100,
        clientY: 100
      }),
      preventDefault: jest.fn(),
      shiftKey: true,
      stopPropagation: jest.fn()
    } as unknown) as React.MouseEvent;

    // render with undefined
    await waitFor(() => {
      qcContextMenu(mouseEvent, undefined);
      container.rerender(<Display />);
    });
    await waitFor(() => {
      expect(container.baseElement).toMatchSnapshot();
    });

    // render with empty content
    await waitFor(() => {
      qcContextMenu(mouseEvent, undefined);
      container.rerender(<Display />);
    });
    await waitFor(() => {
      expect(container.baseElement).toMatchSnapshot();
    });

    // render with one QC Segment
    await waitFor(() => {
      qcContextMenu(mouseEvent, qcSegment);
      container.rerender(<Display />);
    });
    await waitFor(() => {
      expect(container.baseElement).toMatchSnapshot();
    });
  });
});
