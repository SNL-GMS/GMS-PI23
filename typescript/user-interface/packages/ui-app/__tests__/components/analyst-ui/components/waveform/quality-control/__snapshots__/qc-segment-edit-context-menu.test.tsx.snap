// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QC Segment Edit Context Menu renders QcSegmentContextMenu 1`] = `
<div>
  <div
    class="bp4-context-menu2"
  >
    <div />
  </div>
</div>
`;

exports[`QC Segment Edit Context Menu renders QcSegmentContextMenu 2`] = `
<body
  class="bp4-overlay-open"
>
  <div>
    <div
      class="bp4-context-menu2"
    >
      <div />
    </div>
  </div>
  <div
    class="bp4-portal"
  >
    <div
      class="bp4-context-menu2-virtual-target"
      style="left: 100px; top: 100px;"
    />
  </div>
  <div
    class="bp4-portal"
  >
    <div
      aria-live="polite"
      class="bp4-overlay bp4-overlay-open"
    >
      <div
        class="bp4-overlay-start-focus-trap bp4-popover2-appear bp4-popover2-appear-active"
        tabindex="0"
      />
      <div
        class="bp4-overlay-backdrop bp4-popover2-backdrop bp4-context-menu2-backdrop bp4-popover2-appear bp4-popover2-appear-active"
      />
      <div
        class="bp4-popover2-transition-container bp4-overlay-content bp4-popover2-appear bp4-popover2-appear-active"
        style="position: fixed; left: 0px; top: 0px;"
      >
        <div
          class="bp4-popover2 bp4-minimal bp4-popover2-placement-bottom bp4-context-menu2-popover2"
          style="transform-origin: center top;"
        >
          <div
            class="bp4-popover2-content"
          >
            <div />
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`QC Segment Edit Context Menu renders QcSegmentContextMenu 3`] = `
<body
  class="bp4-overlay-open"
>
  <div>
    <div
      class="bp4-context-menu2"
    >
      <div />
    </div>
  </div>
  <div
    class="bp4-portal"
  >
    <div
      class="bp4-context-menu2-virtual-target"
      style="left: 100px; top: 100px;"
    />
  </div>
  <div
    class="bp4-portal"
  >
    <div
      aria-live="polite"
      class="bp4-overlay bp4-overlay-open"
    >
      <div
        class="bp4-overlay-start-focus-trap bp4-popover2-appear bp4-popover2-appear-active"
        tabindex="0"
      />
      <div
        class="bp4-overlay-backdrop bp4-popover2-backdrop bp4-context-menu2-backdrop bp4-popover2-appear bp4-popover2-appear-active"
      />
      <div
        class="bp4-popover2-transition-container bp4-overlay-content bp4-popover2-appear bp4-popover2-appear-active"
        style="position: fixed; left: 0px; top: 0px;"
      >
        <div
          class="bp4-popover2 bp4-minimal bp4-popover2-reference-hidden bp4-popover2-popper-escaped bp4-popover2-placement-right bp4-context-menu2-popover2"
          style="transform-origin: left center;"
        >
          <div
            class="bp4-popover2-content"
          >
            <div />
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`QC Segment Edit Context Menu renders QcSegmentContextMenu 4`] = `
<body
  class=""
>
  <div>
    <div
      class="bp4-context-menu2"
    >
      <div />
    </div>
  </div>
  <div
    class="bp4-portal"
  >
    <div
      class="bp4-context-menu2-virtual-target"
    />
  </div>
</body>
`;

exports[`QC Segment Edit Context Menu renders QcSegmentEditContextMenuContent 1`] = `<div />`;

exports[`QC Segment Edit Context Menu renders QcSegmentEditContextMenuContent 2`] = `
<div>
  <div
    class="form qc-segment-form"
  >
    <div
      class="form__header"
    >
      <div>
        View QC Segment
      </div>
      <div
        class="form__header-decoration"
      >
        <div>
          <div
            class="qc-segment-swatch"
            style="background-color: rgb(235, 6, 200);"
          />
          <span
            class="qc-segment-swatch-label"
          >
            Analyst Defined
          </span>
        </div>
      </div>
    </div>
    <div
      class="form"
    >
      <div
        class="form__panel-selector"
      >
        <div
          class="bp4-button-group"
        >
          <button
            class="bp4-button bp4-active"
            type="button"
            value="Current Version"
          >
            <span
              class="bp4-button-text"
            >
              <span
                class="bp4-button-text"
              >
                Current Version
              </span>
            </span>
          </button>
          <button
            class="bp4-button"
            type="button"
            value="All Versions"
          >
            <span
              class="bp4-button-text"
            >
              <span
                class="bp4-button-text"
              >
                All Versions
              </span>
            </span>
          </button>
        </div>
      </div>
      <div
        class="form-body"
      >
        <div
          class="form-label"
        >
          Category
        </div>
        <div
          class="form-value form-value--uneditable monospace"
          title=""
        >
          Analyst Defined
        </div>
        <div
          class="form-label false"
        >
          Type
        </div>
        <div
          class="form-value form-value--uneditable monospace"
          title=""
        >
          Aggregate
        </div>
        <div
          class="form-label"
        >
          Channel name
        </div>
        <div
          class="form-value form-value--uneditable monospace"
          title=""
        >
          <div
            class="form__text--wrap"
          >
            PDAR.PD01.SHZ
          </div>
        </div>
        <div
          class="form-label"
        >
          Start time
        </div>
        <div
          class="form-value form-value--uneditable monospace"
          title=""
        >
          1970-01-01 00:00:00.000
        </div>
        <div
          class="form-label"
        >
          End time
        </div>
        <div
          class="form-value form-value form-value--time monospace"
          title=""
        >
          1970-01-01 00:01:40.000
        </div>
        <div
          class="form-label"
        >
          Stage
        </div>
        <div
          class="form-value form-value--uneditable monospace"
          title=""
        >
          Unknown
        </div>
        <div
          class="form-label"
        >
          Author
        </div>
        <div
          class="form-value form-value--uneditable monospace"
          title=""
        >
          User 1
        </div>
        <div
          class="form-label"
        >
          Effective at
        </div>
        <div
          class="form-value form-value--uneditable monospace"
          title=""
        >
          1970-01-01 00:00:00.000
        </div>
        <div
          class="form-label"
        >
          Rationale
        </div>
        <div
          class="form-value form-value--uneditable"
          title=""
        >
          <div>
            <textarea
              class="form__text-input"
              data-cy="textarea"
              rows="4"
              title="Rationale"
            />
          </div>
        </div>
      </div>
      <div
        class="form__buttons"
      >
        <div
          class="form__buttons--right"
        >
          <button
            class="bp4-button"
            type="button"
          >
            <span
              aria-hidden="true"
              class="bp4-icon bp4-icon-edit"
              icon="edit"
              tabindex="-1"
            >
              <svg
                data-icon="edit"
                height="16"
                viewBox="0 0 16 16"
                width="16"
              >
                <path
                  d="M3.25 10.26l2.47 2.47 6.69-6.69-2.46-2.48-6.7 6.7zM.99 14.99l3.86-1.39-2.46-2.44-1.4 3.83zm12.25-14c-.48 0-.92.2-1.24.51l-1.44 1.44 2.47 2.47 1.44-1.44c.32-.32.51-.75.51-1.24.01-.95-.77-1.74-1.74-1.74z"
                  fill-rule="evenodd"
                />
              </svg>
            </span>
            <span
              class="bp4-button-text"
            >
              Modify
            </span>
          </button>
          <button
            class="bp4-button bp4-button form__button"
            type="submit"
          >
            <span
              class="bp4-button-text"
            >
              <span
                class="bp4-button-text"
              >
                Cancel
              </span>
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
